hdl21/tests/test_prefix.py

```diff
@@ -86,76 +93,287 @@ def test_prefix_from_exp():
     assert h.Prefix.from_exp(21) == h.Prefix.ZETTA
     assert h.Prefix.from_exp(24) == h.Prefix.YOTTA
 
-    # Check a few unsupported values
-    assert h.Prefix.from_exp(-100) is None
-    assert h.Prefix.from_exp(+100) is None
-
 
 def test_prefix_mul():
     """Test `Prefix` multiplication."""
-    from hdl21.prefix import µ
+    from hdl21.prefix import µ, M, e
 
     assert 5 * µ == 5 * h.Prefix.MICRO
     assert 5 * µ == h.Prefixed(5, h.Prefix.MICRO)
+    assert µ * M == e(0)
+
+
+def test_prefix_div():
+    """Test `Prefix` division"""
+    from hdl21.prefix import µ, M, e
+
+    assert M / µ == e(12)
+    assert µ / M == e(-12)
+
+
+def test_prefix_pow():
+    """Test `Prefix` exponentiation"""
+    from hdl21.prefix import µ, e
+
+    assert µ**2 == e(-12)
```

Comment:
Sorry if I'm just missing it, but how does the equality here work? I don't see a `__eq__` method for either `Exponent` or `Prefixed` that's aware of each other. Is the answer that the `_scale_to_smaller` and `round` methods take either?

Reply:
The logic is that a power operator on `Prefix` that returns a `Prefix` isn't well-defined mathematically, what we'd expect doing this by hand is, e.g.

```
c ** 2 = 0.1 m | K ** 0.37 = ~1.28 UNIT | etc.
```

So, the power operator on `Prefix` returns an `Exponent` instance which can then be compared with `e()`, which also returns an `Exponent`, using Python's native object equality.

The idea in general, is that `Exponent` is a "virtual" class that allows `Prefix` arithmetic (ie. `Prefix-Prefix` multiplication and `Prefix-Real` exponentiation`) to be arbitrary.